pyproject.toml

```diff
@@ -1,7 +1,50 @@
 [build-system]
 requires = [
-    "setuptools>=42",
-    "wheel"
+    "hatchling>=0.21.0",
 ]
+build-backend = "hatchling.build"
 
-build-backend = "setuptools.build_meta"
+[project]
+name = "pymdown-extensions"
+description = "Extension pack for Python Markdown."
+readme = "README.md"
+license = "MIT"
+requires-python = ">=3.7"
+authors = [
+    { name = "Isaac Muse", email = "Isaac.Muse@gmail.com" },
+]
+keywords = [
+    "extensions",
+    "markdown",
+]
+dynamic = [
+    "classifiers",
+    "dependencies",
+    "version",
+]
+
+[project.urls]
+Homepage = "https://github.com/facelessuser/pymdown-extensions"
+
+[tool.hatch.version]
+source = "code"
+path = "pymdownx/__meta__.py"
+
+[tool.hatch.build.targets.sdist]
+include = [
```

Comment:
NOTE TO SELF: not sure if the includes are all correct. Will need to double-check that we are including what we expect in the sdist and not what we are don't.